doc/Customization-guide/Code-style/Objective-C/index.md

```diff
@@ -4,4 +4,283 @@ title: Objective-C
+
+Don’t use line breaks in method declarations.
+
+Prefer exposing an immutable type for a property if it being mutable is an implementation detail. This is a valid reason to declare an `ivar` for a property.
+
+Declare properties readonly if they are only set once in `-init`.
+
+Declare properties copy if they return immutable objects and aren't ever mutated in the implementation.
+
+Don't use `@synthesize` unless the compiler requires it. Note that optional properties in protocols must be explicitly synthesized in order to exist.
+
+Instance variables should be prefixed with an underscore (just like when implicitly synthesized).
+
+Class initializers should generally return `instancetype` rather than `id`.
+
+# Expressions
+
+Don't access an `ivar` unless you're in `-init` or `-dealloc`, custom setters, getters or tight loops.
+
+Use dot syntax when accessing property.
+
+Use object literals, boxed expressions, and subscripting over the older alternatives.
+
+```objc
+NSNumber *boolYES = @YES;
+NSMutableArray *instrumentsMutable = [@[ @"Ocarina", @"Flute", @"Harp"] mutableCopy];
+NSDictionary *stats = @{@"Character": @"Zelda", @"Weapon": @"Sword", @"Hitpoints": @50};
+NSString *name = stats[@"Character"];
+NSNumber *diameter = @( 2 * pi * r );
+```
+
+Comparisons should be explicit for everything except `BOOL`s.
+
+One space between operators and operands.
+
+```objc
+NSInteger value = this + that;
+```
+
+Prefer positive comparisons to negative.
+
+Short form, `nil` coalescing ternary operators should avoid parentheses.
+
+```objc
+Blah *a = thingThatCouldBeNil ?: defaultValue;
+```
+
+There shouldn't be a space between a cast and the variable being cast.
+
+```objc
+NewType a = (NewType)b;
+```
+
+Pointer notation should be applied to the variable and not the type
+
+```objc
+NSInteger *counter = 0;
+```
+
+# Whitespace
+
+Curly braces general rule - place them on the same line before and with a newline after.
+
+Add a space after the `-` or `+` in the method declaration.
+
+Space between type and `*` in return type and argument type.
+
+No space between the closing paren of the return type and the method name.
+
+No space between the closing paren of the argument type and the argument name.
+
+No tabs or spaces at the end of lines, including empty lines.
+
+Add a space after the type but not the variable name (e.g, `NSString *string`).
+
+End files with a newline.
+
+Make liberal use of vertical whitespace to divide code into logical chunks.
+
+Space between `//` and the first character of a comment.
+
+Spaces for indentation and for alignment within a line, not tabs.
+
+Always surround `if` and `else` bodies with curly braces.
+
+`return` and `break` early at the top of methods or in obvious places.
+Try not to place early `return`s in the middle of functions as the control flow is then not clear (you can create `result` variable, which will get it’s value somewhere in the method and return it).
```

Comment:
As I understand "result" isn't a variable name or statement in this context, so you don't have to mark it.